Mathlib/LinearAlgebra/Basis.lean

```diff
@@ -1393,6 +1393,19 @@ def Submodule.inductionOnRankAux (b : Basis ι R M) (P : Submodule R M → Sort*
 
 end Induction
 
+/-- An element of a non-unital-non-associative algebra is in the center exactly when it commutes
+with the basis elements. -/
+lemma Basis.mem_center_iff {A}
+    [Semiring R] [NonUnitalNonAssocSemiring A]
+    [Module R A] [SMulCommClass R A A] [IsScalarTower R A A]
+    (b : Basis ι R A) {x : A} :
+    x ∈ Set.center A ↔ ∀ i, Commute (b i) x := by
+  constructor
+  · intro h i; apply h
+  · intro h y
+    rw [← b.total_repr y, Finsupp.total_apply, Finsupp.sum, Finset.sum_mul, Finset.mul_sum]
+    simp_rw [mul_smul_comm, smul_mul_assoc, (h _).eq]
```

Comment:
Way out of scope for this PR but: we should really have a nicer way to say that each element can be written as a sum of scaled basis vectors. Maybe a theorem along the lines of `Basis.induction` saying that `P` is always true on `A` whenever it is true on each `b i` and when it's closed under addition and scalar multiplication?